osu.Framework/Platform/Storage.cs

```diff
@@ -98,6 +98,23 @@ public virtual Storage GetStorageForDirectory(string path)
             return (Storage)Activator.CreateInstance(GetType(), fullPath);
         }
 
+        /// <summary>
+        /// Move a file from one location to another. File must exist. Destination must not exist.
+        /// </summary>
+        /// <param name="from">The file path to move.</param>
+        /// <param name="to">The destination path.</param>
+        public abstract void Move(string from, string to);
+
+        /// <summary>
+        /// Create a new file on disk, using a temporary file to write to before moving to the final location to ensure a half-written file cannot exist at the specified location.
+        /// </summary>
+        /// <remarks>
+        /// If the target file path already exists, it will be deleted before attempting to write a new version.
+        /// </remarks>
+        /// <param name="path">The path of the file to create or overwrite.</param>
+        /// <returns></returns>
```

Comment:
```suggestion
        /// <returns>A <see cref="Stream"/> to which contents of the file to be created safely should be written.</returns>
```

or just remove, idunno - bit hard to explain that one